RestingStateStats/RestingStateStats.sh

```diff
@@ -993,7 +992,7 @@ main()
 			
 			# TBD: change these paths to use variables instead of hard coded paths
 			touch ${matlab_script_file_name}
-			echo "addpath ${HCPPIPEDIR}/RestingStateStats " >> ${matlab_script_file_name}
+			echo "addpath ${HCPPIPEDIR}/scripts/RestingStateStats " >> ${matlab_script_file_name}
 			echo "addpath /home/HCPpipeline/pipeline_tools/gifti" >> ${matlab_script_file_name}
```

Comment:
We should fix this line to not call an irrelevant hard-coded path, but rather `addpath ${HCPPIPEDIR}/global/matlab`.  Note that because I previously added a symlink there of `@gifti@ -> gifti-1.6/@gifti`, adding `${HCPPIPEDIR}/global/matlab` to the matlab path has the effect of automatically adding both the cifti{open,save,savereset}.m functions to the matlab path and their needed gifti support as well (the latter being the intended purpose of L996).

Reply:
Also, L995 is wrong.  It should be `addpath ${HCPPIPEDIR}/RestingStateStats/scripts`, or use something like the `addpath('${this_script_dir}/scripts')` construction that we've added to the other matlab-related scripts, to make it more robust against directory name changes.

Reply:
* I've adapted `RestingStateStats.sh` to use similar techniques for invoking MATLAB or Octave to the other scripts.
* I also added comments to `Compile_All_MATLAB_code.sh` noting that users really should not need to do this.
* I fixed the use of `addpath` statements in `RestingStateStats.sh` to no longer contain a hard coded path to the gifti code
* I fixed the mistaken use of `${HCPPIPEDIR}/scripts/RestingStateStats` to `${HCPPIPEDIR}/RestingStateStats/scripts` in the interpreted code section

* I'm unclear on whether there has been a decision that we should replace this RestingStateStats code with code from the `rss-mrfix` branch or whether the code in the `rss-mrfix` branch should be merged into this branch with different file names so that it will end up added to the `master` branch when this pull request is merged, but still have separate file names (So both this code and the code from the `rss-mrfix` branch will co-exist.)

Reply:
Probably need to stay separate until the code is actually merged.